osf/management/commands/force_archive.py

```diff
@@ -0,0 +1,344 @@
+from django.db.utils import IntegrityError
+from django.utils import timezone
+
+from osf.models import Registration
+from addons.osfstorage.models import OsfStorageFile, OsfStorageFolder
+from scripts import utils as script_utils
+from website.archiver import ARCHIVER_SUCCESS
+from website.settings import ARCHIVE_TIMEOUT_TIMEDELTA
+
+logger = logging.getLogger(__name__)
+
+# Logging globals
+CHECKED_OKAY = []
+CHECKED_STUCK_RECOVERABLE = []
+CHECKED_STUCK_BROKEN = []
+VERIFIED = []
+SKIPPED = []
+
+# Ignorable NodeLogs
+LOG_WHITELIST = {
+    'affiliated_institution_added',
+    'comment_added',
+    'comment_removed',
+    'comment_restored',
+    'comment_updated',
+    'contributor_added',
+    'contributor_removed',
+    'contributors_reordered',
+    'edit_description',
+    'edit_title',
+    'embargo_approved',
+    'embargo_cancelled',
+    'embargo_completed',
+    'embargo_initiated',
+    'embargo_terminated',
+    'file_tag_added',
+    'license_changed',
+    'made_contributor_invisible',
+    'made_private',
+    'made_public',
+    'made_wiki_private',
+    'made_wiki_public',
+    'permissions_updated',
+    'pointer_created',
+    'pointer_removed',
+    'prereg_registration_initiated',
+    'project_registered',
+    'registration_approved',
+    'registration_cancelled',
+    'registration_initiated',
+    'retraction_approved',
+    'retraction_initiated',
+    'tag_added',
+    'tag_removed',
+    'wiki_deleted',
+    'wiki_updated'
+}
+
+# Require action, but recoverable from
+LOG_GREYLIST = {
+    'osf_storage_file_added',
+    'osf_storage_file_removed',
+    'osf_storage_file_updated',
+    'osf_storage_folder_created'
+}
+
+def complete_registration(reg):
+    archive_job = reg.archive_job
+    target = archive_job.get_target('osfstorage')
+    target.status = ARCHIVER_SUCCESS
+    target.save()
+    archive_job._post_update_target()
+
+def manually_archive(tree, reg, parent=None):
+    if not isinstance(tree, list):
+        tree = [tree]
+    for filenode in tree:
+        if filenode['deleted']:
+            continue
+        file_obj = filenode['object']
+        cloned = file_obj.clone()
+        if cloned.is_deleted:
+            if cloned.is_file:
+                cloned.recast(OsfStorageFile._typedmodels_type)
+            else:
+                cloned.recast(OsfStorageFolder._typedmodels_type)
+        if not parent:
+            parent = reg.get_addon('osfstorage').get_root()
+            cloned.name = 'Archive of OSF Storage'
+        cloned.parent = parent
+        cloned.node = reg
+        cloned.copied_from = file_obj
+        try:
+            cloned.save()
+        except IntegrityError:
+            # Files sometimes already created
+            cloned = reg.files.get(name=cloned.name)
+
+        if file_obj.versions.exists() and filenode['version']:  # Min version identifier is 1
+            if not cloned.versions.filter(identifier=filenode['version']).exists():
```

Comment:
Only in the event that a file copy is successful, but the completion of the archive task fails will this be avoided.